components-core/src/main/java/com/adyen/checkout/components/base/BaseConfigurationBuilder.kt

```diff
@@ -63,6 +66,21 @@ abstract class BaseConfigurationBuilder<
         return this as BuilderT
     }
 
+    /**
+     * TODO docs
+     */
+    fun setAmount(amount: Amount): BuilderT {
```

Comment:
Does calling `setAmount(Amount.EMPTY)` trigger an exception? If so should we allow it maybe?

Reply:
It will. We didn't use to allow it though. For what use case we should allow it?

Reply:
Not sure, just thinking that `Amount.EMPTY` is a valid option, so if you have a case where your `amount` might be EMPTY sometimes or positive other times, you'll have to change this:
```kotlin
return DropInConfiguration.Builder(
    shopperLocale,
    environment,
    clientKey,
)
.addCardConfiguration(cardConfiguration)
.addBcmcConfiguration(bcmcConfiguration)
.addGooglePayConfiguration(googlePayConfiguration)
.add3ds2ActionConfiguration(threeDS2Configuration)
.setAmount(amount)
.build()
```
to this:
```kotlin
val dropInConfigurationBuilder = DropInConfiguration.Builder(
    shopperLocale,
    environment,
    clientKey,
)
.addCardConfiguration(cardConfiguration)
.addBcmcConfiguration(bcmcConfiguration)
.addGooglePayConfiguration(googlePayConfiguration)
.add3ds2ActionConfiguration(threeDS2Configuration)

if(amount != Amount.EMPTY) {
    dropInConfigurationBuilder.setAmount(amount)
}

return dropInConfigurationBuilder.build()
```
Which is not terrible but not nice either. Does that make any sense?